user-service/services.js

```diff
@@ -13,5 +18,37 @@ export async function hashSaltPassword(password) {
  * @param {String} hashed taken from DB
  */
 export async function verifyPassword(password, hashed) {
-    return bcrpyt.compare(password, hashed);
+    return bcrypt.compare(password, hashed);
+}
+
+export async function generateAccessToken(user) {
+    return jwt.sign(
+        { username: user.username },
+        process.env.ACCESS_TOKEN_SECRET,
+        { expiresIn: '30s' },
+    ); // expire in 30s
+}
+
+export async function generateRefreshAccessToken(user) {
+    return jwt.sign(
+        { username: user.username },
+        process.env.REFRESH_TOKEN_SECRET,
+        { expiresIn: '1800s' },
+    ); // expire in 30 mins
+}
+
+export async function verifyAccessToken(token) {
+    return jwt.verify(token, process.env.ACCESS_TOKEN_SECRET, (err, user) => {
+        console.log(err);
+        if (err) return undefined;
+        return user;
+    });
+}
+
+export async function verifyRefreshToken(refreshToken) {
+    return jwt.verify(refreshToken, process.env.REFRESH_TOKEN_SECRET, async (err, user) => {
+        if (err) return undefined;
+        const token = await generateAccessToken({ username: user.username });
+        return token;
```

Comment:
I think you should handle the possible errors and call these functions in a try-catch block in the controller. Can raise an issue for this